### PR TITLE
[Feature] [connectors] Add spark cassandra sink

### PR DESCRIPTION
**Pull request by zixi0825**
## Purpose of this pull request

This pull request adds spark cassandra sink. #985

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by zixi0825** [comment]
@RickyHuo @CalvinKirs

**Comment by CalvinKirs** [comment]
hi, because I'm dealing with a license issue recently, the reply is relatively late, you need to update your code.

File: seatunnel-connectors/seatunnel-connector-spark-cassandra/src/main/scala/org/apache/seatunnel/spark/sink/Cassandra.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        writer.options(optionMap)
+      case Failure(exception) => // do nothing
+    }
+
+    writer.save()
+  }
+
+  override def checkConfig(): CheckResult = {
+    check(config, "table", "keyspace")
+  }
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-cassandra/src/main/scala/org/apache/seatunnel/spark/sink/Cassandra.scala:59]: We have now changed `check` to `checkAllExists` , please check, thanks

File: docs/en/spark/configuration/sink-plugins/Cassandra.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Save mode
+
+### Example
+
+```bash
+cassandra {
+    table = "t2"
+    keyspace = "excelsior"
+    result_table_name = "test"
+}
```
Comment (wuchunfu) [inline docs/en/spark/configuration/sink-plugins/Cassandra.md:57]: `sink` does not need `result_table_name` , please check, thanks

**Comment by CalvinKirs** [comment]
I closed this PR due to long time no response, please ping me if there is an update.

**Comment by zixi0825** [comment]
@CalvinKirs please reopen it ,thanks

**Pull request closed without merging**